pkg/s3select/input.go

```diff
@@ -151,7 +151,7 @@ func NewInput(opts *Options) (*Input, error) {
 	}
 	reader.firstRow = nil
 
-	reader.reader.FieldsPerRecord = -1
+	reader.reader.FieldsPerRecord = 0
```

Comment:
This would be a bug when csv files have different numbers of fields in each line.

Reply:
I think S3 is not supporting that AFAIK

Reply:
Then I think we can make sure that if Header is set we don't need to honor variadic records.